server/aws-lsp-partiql/src/server/hover-info/parser-type.ts

```diff
@@ -0,0 +1,46 @@
+import Parser, { SyntaxNode } from 'web-tree-sitter'
+import { initParser } from '../tree-sitter-parser/parserUtil'
+import { Hover, MarkupKind } from '@aws/language-server-runtimes/server-interface'
+import { hoverDictionary } from './hoverDictionary'
+
+// Global or service-level variables
+let parserInitialized = false
+let globalParser: Parser | null = null
+
+async function ensureParserInitialized() {
+    if (!parserInitialized) {
+        globalParser = await initParser()
```

Comment:
This would create a new parser instance separate from the one we use for the semantic tokens right? Do you see a way we could use the single instance in both places?

Reply:
You are right. There is no need to use separate parsers. Will put the globalParser into `parserUtil.ts`.